src/Board.tsx

```diff
@@ -1,34 +1,21 @@
-import * as React from 'react';
+import React, { FunctionComponent } from 'react';
 import { Square } from './Square';
 
 type Props = {
   squares: string[];
+  clickHandler: Function;
```

Comment:
type sig

Reply:
So a little bit of :rubber-ducking: here for the sake of relating a learning experience.

1) The `onClick` prop has a sig of 

```ts
(event: MouseEvent<HTMLButtonElement, MouseEvent>) => void
```

2) I wasn't sure what exactly to specify as the `clickHandler` sig to make it compatible with the above. I tried the above explicitly like so:

```ts
type Props = {
  value: string;
  clickHandler: (event: MouseEvent<HTMLButtonElement, MouseEvent>) => void;
};
```

but that generated a tsc error: `Type 'MouseEvent' is not generic. ts(2315) [5,25]`

3) Googling that error, I found an SO post asking [**Why is the `MouseEvent` in the checkbox event handler not generic?**](https://stackoverflow.com/q/44764146/70876). The [answer](https://stackoverflow.com/a/44764395/70876) suggested using 

```ts
  (event: React.MouseEvent<HTMLInputElement>) => void;
```

which still resulted in an error because I'm using a `<button>` and not an `<input>`. 

4) So this is the solution:

```ts
  clickHandler: (event: React.MouseEvent<HTMLButtonElement>) => void;
```

### Moral of the Story

I never would have figured that out without google and stackoverflow. 🤦‍♂

Reply:
And actually, this story belongs to `Square.tsx`. The sig for `Board`'s copy of the `clickHandler` is different.

Reply:
You can also say `or` for the generics—like `React.MouseEvent<HTMLButtonElement | HTMLInputElement >`

Reply:
but yep! Nice story walk through